nordicsemi/dfu/dfu_transport_serial.py

```diff
@@ -445,31 +448,32 @@ def __get_checksum_response(self):
 
     def __stream_data(self, data, crc=0, offset=0):
         logger.debug("Serial: Streaming Data: " +
-            "len:{0} offset:{1} crc:0x{2:08X}".format(len(data), offset, crc))
+                     "len:{0} offset:{1} crc:0x{2:08X}".format(len(data), offset, crc))
+
         def validate_crc():
             if (crc != response['crc']):
-                raise ValidationException('Failed CRC validation.\n'\
-                                + 'Expected: {} Recieved: {}.'.format(crc, response['crc']))
+                raise ValidationException('Failed CRC validation.\n'
+                                          + 'Expected: {} Recieved: {}.'.format(crc, response['crc']))
             if (offset != response['offset']):
-                raise ValidationException('Failed offset validation.\n'\
-                                + 'Expected: {} Recieved: {}.'.format(offset, response['offset']))
+                raise ValidationException('Failed offset validation.\n'
+                                          + 'Expected: {} Recieved: {}.'.format(offset, response['offset']))
 
-        current_pnr     = 0
+        current_pnr = 0
 
-        for i in range(0, len(data), (self.mtu-1)/2 - 1):
+        for i in range(0, len(data), int((self.mtu-1)/2 - 1)):
             # append the write data opcode to the front
             # here the maximum data size is self.mtu/2,
             # due to the slip encoding which at maximum doubles the size
-            to_transmit = data[i:i + (self.mtu-1)/2 - 1 ]
-            to_transmit = struct.pack('B',DfuTransportSerial.OP_CODE['WriteObject']) + to_transmit
+            to_transmit = data[i:i + int((self.mtu-1)/2 - 1)]
```

Comment:
This might not fix the issue if `self.mtu` is float the result of `//` will be a float too and throw a TypeError exception.
E.g. `1.0 // 2 = 0.1 ; 1 // 2 = 0`

Reply:
The problem with allowing `self.mtu` to be float is that you are weakening type/value safety for other users of this library. MTU values correspond to lengths of byte arrays. That forces them conceptually to be integers. Python makes a distinction between integers and floating points, and we should use that. We could conceptually accept floats with zero fractional part and throw an error when the criterion doesn't match. But this criterion can be ensured by using types correctly instead, and less code is better code. And I don't see why a user of this library would want to represent MTU values as floats, so I see no counterargument appealing to usability.

Side note: Ideally the type would be something like positive integer no larger than what the softdevice supports. But the unfamiliarity of any nonstandard types we would have to introduce makes this very unpythonic.